### PR TITLE
ErrorCode and ErrorKind columns to executions

### PR DESCRIPTION
**Pull request by kumare3**
# TL;DR
The intention is to make it possible to calculate errors for executions or node_executions by kind and eventually categorize by error codes

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [x] Unit tests added
 - [x] Code documentation added
 - [x] Any pending items have an associated Issue

## Complete description
Refer to issue

## Tracking Issue
https://github.com/lyft/flyte/issues/300

## Follow-up issue
NA

### CODE REVIEW

File: pkg/repositories/config/migrations.go [hunk 1]
```diff
@@ -199,4 +199,14 @@ var Migrations = []*gormigrate.Migration{
 			return tx.Exec("ALTER TABLE workflows ADD COLUMN IF NOT EXISTS state integer;").Error
 		},
 	},
+	// Modify the executions & node_executison table, if necessary
+	{
+		ID: "2020-04-29-executions",
+		Migrate: func(tx *gorm.DB) error {
+			return tx.Exec("DO we need to do migrations for nullable column adds?").Error
```
Comment (katrogan) [inline pkg/repositories/config/migrations.go:206]: yep, we still need to add the schema to the table - without this the most recent version of the executions model will determine the table schema

File: pkg/repositories/transformers/execution.go [hunk 2]
```diff
@@ -135,6 +136,10 @@ func UpdateExecutionModelState(
 				},
 			},
 		}
+	} else if request.Event.GetError() != nil {
```
Comment (EngHabu) [inline pkg/repositories/transformers/execution.go:139]: nit: I think this should move to the next if block

Comment (katrogan) [inline pkg/repositories/transformers/execution.go:139]: isn't this condition checked in line 145 below? can you merge the blocks?

File: pkg/repositories/config/migrations.go (hunk 1 above)
Comment (kumare3) [reply pkg/repositories/config/migrations.go:206]: so alter table?

Comment (kumare3) [reply pkg/repositories/config/migrations.go:206]: and how about indexes? This sounds a little weird. Can we just not run auto-migration on the model?

File: pkg/repositories/transformers/execution.go (hunk 2 above)
Comment (kumare3) [reply pkg/repositories/transformers/execution.go:139]: done

Comment (kumare3) [reply pkg/repositories/transformers/execution.go:139]: done

File: pkg/repositories/models/execution.go [hunk 3]
```diff
@@ -51,4 +51,8 @@ type Execution struct {
 	InputsURI storage.DataReference
 	// User specified inputs. This map might be incomplete and not include defaults applied
 	UserInputsURI storage.DataReference
+	// Execution Error Kind. nullable
```
[the remark this answers is not in the document]
Comment (kumare3) [reply pkg/repositories/models/execution.go:54]: yes, is there a way to document that, reference to flyteidl?

[the remark this answers is not in the document]
Comment (katrogan) [reply pkg/repositories/models/execution.go:54]: that would be great if you don't mind (sorry I realized it's missing from Phase above)

[the remark this answers is not in the document]
Comment (kumare3) [reply pkg/repositories/models/execution.go:54]: done

**Pull request merged**